### PR TITLE
[SofaCUDA,SofaDistanceGrid] Move DistanceGrid CUDA files in a SofaDistanceGrid extension

### PR DESCRIPTION
**Pull request by alxbilger**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/SofaDistanceGrid/extensions/CUDA/src/SofaDistanceGrid/CUDA/CudaDistanceGridContactMapper.cu [hunk 1]
```diff
@@ -21,10 +21,16 @@
 ******************************************************************************/
 #include <vector_types.h>
 #include <sofa/gpu/cuda/mycuda.h>
+#include "cuda.h"
 
-
-namespace sofa::gpu::cuda
+#if defined(__cplusplus) && CUDA_VERSION < 2000
```
Comment (damienmarchal) [inline applications/plugins/SofaDistanceGrid/extensions/CUDA/src/SofaDistanceGrid/CUDA/CudaDistanceGridContactMapper.cu:26]: why ?

Comment (alxbilger) [reply applications/plugins/SofaDistanceGrid/extensions/CUDA/src/SofaDistanceGrid/CUDA/CudaDistanceGridContactMapper.cu:26]: I also would like to know. NVCC is still a mystery for me...

**Comment by bakpaul** [comment]
Compilation issue on Windows. Modernize the CUDA Cmake before this PR. This plugin adds C source file, making it harder than other reordering done before. Problem arise from the fact that CMake passes compile option to NVCC that the compiler doesn't understand.

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Review by fredroy — APPROVED** [review]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

**Pull request merged**